src/connectedk8s/azext_connectedk8s/_client_factory.py

```diff
@@ -30,36 +50,74 @@ def cf_connected_cluster_prev_2022_05_01(cli_ctx, _):
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        return ResourceManagementClient(credential=credential, subscription_id=os.getenv('AZURE_SUBSCRIPTION_ID'),
+                                        base_url=cli_ctx.cloud.endpoints.resource_manager, credential_scopes=[cli_ctx.cloud.endpoints.resource_manager + '/.default'])
     return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id)
 
 
-def _resource_providers_client(cli_ctx):
-    from azure.mgmt.resource import ResourceManagementClient
-    return get_mgmt_service_client(cli_ctx, ResourceManagementClient).providers
+def resource_providers_client(cli_ctx, subscription_id=None):
+    return _resource_client_factory(cli_ctx, subscription_id).providers
 
     # Alternate: This should also work
     # subscription_id = get_subscription_id(cli_ctx)
     # return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id).providers
 
 
 def _graph_client_factory(cli_ctx, **_):
-    profile = Profile(cli_ctx=cli_ctx)
-    cred, _, tenant_id = profile.get_login_credentials(
-        resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-    client = GraphRbacManagementClient(cred, tenant_id,
-                                       base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    if os.getenv('AZURE_ACCESS_TOKEN'):
+        credential = AccessTokenCredential(access_token=os.getenv('AZURE_ACCESS_TOKEN'))
+        client = GraphRbacManagementClient(credential, os.getenv('AZURE_TENANT_ID'),
+                                           base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    else:
+        profile = Profile(cli_ctx=cli_ctx)
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+        client = GraphRbacManagementClient(cred, tenant_id,
+                                           base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
     configure_common_settings(cli_ctx, client)
     return client
 
 
 def get_graph_client_service_principals(cli_ctx):
     return _graph_client_factory(cli_ctx).service_principals
+
+
+class AccessTokenCredential:
+    """Simple access token Authentication. Returns the access token as-is.
+    """
+
+    def __init__(self, access_token):
+        self.access_token = access_token
+
+    def get_token(self, *arg, **kwargs):
+        import time
+        # Assume the access token expires in 60 minutes
+        return AccessToken(self.access_token, int(time.time()) + 3600)
+
+    def signed_session(self, session=None):
+        session = session or requests.Session()
+        header = "{} {}".format('Bearer', self.access_token)
+        session.headers['Authorization'] = header
+        return session
```

Comment:
This is only for Track 1 SDK. Most likely you won't need it. :)

Reply:
I think we require this since the 'GraphRbacManagementClient' still calls this function. When this function was not implemented , I saw the following error : ''AccessTokenCredential' object has no attribute 'signed_session''

Reply:
`GraphRbacManagementClient` is the old Track 1 SDK for AD Graph. Consider using MS Graph client: https://github.com/Azure/azure-cli/blob/dev/doc/microsoft_graph_client.md